states/oklahoma/parse_pdf.py

```diff
@@ -0,0 +1,105 @@
+from common import cache_request
+from io import BytesIO
+from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter
+from pdfminer.converter import TextConverter
+from pdfminer.layout import LAParams
+from pdfminer.pdfpage import PDFPage
+from io import StringIO
+import os
+from collections import defaultdict
+
+
+def get_text(url):
+  req = cache_request(url, is_binary=True)
+  with BytesIO(req) as fh:
+    output = StringIO()
+    manager = PDFResourceManager()
+    converter = TextConverter(manager, output, laparams=LAParams())
+    interpreter = PDFPageInterpreter(manager, converter)
+
+    for page in PDFPage.get_pages(fh, set()):
+      interpreter.process_page(page)
+    
+    converter.close()
+    text = output.getvalue()
+    output.close()
+    return text
+
+
+def group_rows(lines):
+  # There is no email address, and fax and phone share a line.
+  sections = ['COUNTY', 'PHONE', 'MAILING ADDRESS']
+
+  # Only add data to output if it is between one of the above sections and an empty string.
+  add_data = False
+
+  index = 0
+
+  first_page_rows = []
+  second_page_rows = []
+
+  rows = first_page_rows
+
+  for line in lines:
+    # indicates next page has started
+    if 'State of Oklahoma' in line:
+      rows = second_page_rows
+
+    # Edge case where one group lines does not end in empty string.
+    if 'Please call CEB' in line:
+      continue
+
+    if line in sections:
+      add_data = True
+      index = 0
+      continue
+    elif line == '':
+      add_data = False
+
+    if add_data:
+      if index >= len(rows):
+        rows.append([line])
```

Comment:
Instead, make row a list of dictionaries indexed by `'COUNTY', 'PHONE', 'MAILING ADDRESS'`.

You should also `asset` that each of these sections has the same number of items.

Reply:
I misinterpreted this comment initially and did what I have now.  I understand what this comment means now, but I think this way is just as good and makes the assertions a bit more straightforward.